src/averageAges.js

```diff
@@ -1,61 +1,43 @@
 'use strict';
 
-/**
- * Implement calculateMenAverageAge function
- *
- * Function returns average age of men in array. If `century` is specified then
- * function calculates average age only for men who died in this century
- *
- * To calculate century:
- * Divide year of person's death by 100: Math.ceil(person.died / 100)
- *
- * @param {object[]} people
- * @param {number} century - optional
- *
- * @return {number}
- */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const men = people.filter(person => person.sex === 'm');
+
+  const menAges = (century)
+    ? men
+      .filter(person => Math.ceil(person.died / 100) === century)
+      .map(person => person.died - person.born)
+    : men.map((person) => person.died - person.born);
+
+  return (menAges.reduce((sum, age) => sum + age)) / menAges.length;
```

Comment:
create separate function that will take people object and will return average age, to use it here, and in the second function